### ISSUE TITLE
`Encoding.default_external = 'ASCII'` with `StringIO#string` results are different from MRI

### ISSUE THREAD

**Issue by koic** [jruby/jruby#6528]
I found a difference in behavior between MRI and JRuby.

**Environment Information**
  
Provide at least:
* jruby 9.2.14.0 (2.5.7) 2020-12-08 ebe64bafb9 Java HotSpot(TM) 64-Bit Server VM 25.271-b09 on 1.8.0_271-b09 +jit [darwin-x86_64]
* Darwin atelier.local 19.6.0 Darwin Kernel Version 19.6.0: Thu Oct 29 22:56:45 PDT 2020; root:xnu-6153.141.2.2~1/RELEASE_X86_64 x86_64

**Expected Behavior**

The following is an example:

```ruby
# example.rb
require 'stringio'

Encoding.default_external = 'ASCII'

io = StringIO.new
io << '©'

p io.string
p io.string.encoding
```

This is behavior of MRI 2.5.7 (and MRI 3.0.0) .

```console
% ruby -v
ruby 2.5.7p206 (2019-10-01 revision 67816) [x86_64-darwin19]

% ruby example.rb
"\u00A9"
#<Encoding:UTF-8>
```

```console
% ruby -v
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin19]

% ruby example.rb
"\u00A9"
#<Encoding:UTF-8>
```

**Actual Behavior**

This is behavior of JRuby 9.2.14.0.

```console
% ruby -v
jruby 9.2.14.0 (2.5.7) 2020-12-08 ebe64bafb9 Java HotSpot(TM) 64-Bit Server VM 25.271-b09 on 1.8.0_271-b09 +jit [darwin-x86_64]

% ruby example.rb
"\xC2\xA9"
#<Encoding:US-ASCII>
```

I encountered this issue in RuboCop.
https://github.com/rubocop-hq/rubocop/pull/9347

**Comment by headius** [comment]
The behavior of JRuby master (9.3) appears to match:

```
[] ~/projects/jruby $ jruby -v stringio.rb 
jruby 9.3.0.0-SNAPSHOT (2.6.5) 2021-01-05 c51a92ac2e OpenJDK 64-Bit Server VM 16-ea+29-2091 on 16-ea+29-2091 +jit [darwin-x86_64]
stringio.rb:5: warning: setting Encoding.default_external
"\u00A9"
#<Encoding:UTF-8>

[] ~/projects/jruby $ rvm ruby-2.6.5 do ruby -v stringio.rb 
ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-darwin18]
stringio.rb:4: warning: setting Encoding.default_external
"\u00A9"
#<Encoding:UTF-8>
```

I will try to see if it is a trivial fix that can be backported, in case we do a 9.2.15 release.

**Comment by headius** [comment]
Seems like the fix came from https://github.com/jruby/jruby/pull/6183, and since CRuby 2.5 has the same behavior it should be safe for us to backport it. I have opened #6530.

**Comment by headius** [comment]
Fixed by #6530.

**Comment by koic** [comment]
Great! Thank you @headius!
